pkg/rollout/rollout.go

```diff
@@ -0,0 +1,191 @@
+package rollout
+
+import (
+	"fmt"
+
+	runapi "github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/run"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	format "github.com/logrusorgru/aurora"
+	"github.com/pkg/errors"
+	"google.golang.org/api/run/v1"
+)
+
+// Logger describes a logger printing information to the user.
+type Logger interface {
+	Println(args ...interface{})
+	Printf(format string, args ...interface{})
+}
+
+// Rollout is the rollout manager.
+type Rollout struct {
+	RunClient        runapi.Client
+	Config           *config.Config
+	Log              Logger
+	promoteCandidate bool
+}
+
+// Automatic tags.
+const (
+	StableTag    = "stable"
+	CandidateTag = "candidate"
+	LatestTag    = "latest"
+)
+
+// New returns a new rollout manager.
+func New(client runapi.Client, config *config.Config, log Logger) *Rollout {
+	return &Rollout{
+		RunClient:        client,
+		Config:           config,
+		Log:              log,
+		promoteCandidate: false,
+	}
+}
+
+// Manage handles the gradual rollout.
+func (r *Rollout) Manage() (*run.Service, error) {
+	serviceName := generateServiceName(r.Config.Metadata.Project, r.Config.Metadata.Service)
+	svc, err := r.RunClient.Service(serviceName)
```

Comment:
get the project and service name here, don't get API-specific naming convention into this part of code